### DISCUSSION TITLE
NuGet AssemblyFileVersion fixed at 1.0.0.0

### DISCUSSION THREAD

**Discussion by LewisNorton** [TheCodeTraveler/AsyncAwaitBestPractices#85] [Q&A] [answered]
Hi all,

I noticed that the AsyncAwaitBestPractices and AsyncAwaitBestPractices.MVVM packages have their file versions set to 1.0.0.0. This is out of step with the NuGet package version.

Is there a reason for this?

**Background**

We're looking to deploy this dll via an installer, but if the file version doesn't increment then the installer won't update the files. We can work around this but I'd like to get some context around the difference first.

Cheers!

![image](https://user-images.githubusercontent.com/74623/129899760-5d8cb3e2-198f-4743-a493-b6f42615f206.png)

![image](https://user-images.githubusercontent.com/74623/129899831-63935f0c-5641-42b1-a7fc-a354577dd28e.png)

**Comment by TheCodeTraveler** [comment, marked as answer]
Hi @LewisNorton!

I honestly didn't know that the DLL File Version is still used nowadays now that NuGet Packages are the recommended way for adding external libraries.

I'm happy to increment the File Version for you if it'll help 👍 

Just out of curiosity, how come you're using DLL references instead of NuGet?

**Reply by LewisNorton** [reply]
Hi @brminnick 👋 

Thanks for offering to do that, it would really help us.

I should have added some more details in my original post. We do use NuGet to manage our external dependencies. Once the application is built, the output directory contains our assemblies and any dependency assemblies. Here you'll find AsyncAwaitBestPractices.dll and the MVVM variant.

![image](https://user-images.githubusercontent.com/74623/130038684-bd70f5a2-2ad6-416e-baf3-cfafa2e91214.png)

To distribute the application, we create a Windows installer via [WiX](https://wixtoolset.org/). When installing the application, it uses the file version to decide whether to upgrade a file. As the file version is fixed at 1.0.0.0, no upgrade will take place.

This has the possibility to cause runtime errors if a dependent were to call some new part of the library which doesn't exist in the deployed dll.

**Reply by TheCodeTraveler** [reply]
Hi @LewisNorton!

Just FYI - the latest release, v6.0.1, includes the correct File Version 👍 
https://github.com/brminnick/AsyncAwaitBestPractices/releases/tag/v6.0.1
